src/components/forms/riddleUpsertForm/CreateRiddleForm.tsx

```diff
@@ -0,0 +1,114 @@
+import { useForm } from 'react-hook-form-mui';
+import type { ReactNode } from 'react';
+import { useCallback, useState } from 'react';
+import { Box, Step, StepLabel, Stepper } from '@mui/material';
+import { v4 as uuidv4 } from 'uuid';
+import { useNavigate } from 'react-router-dom';
+
+import type { RiddleUpsertDetail } from '../../../utils/Types';
+import { getDifficultyObject } from '../../../utils/Difficulty';
+
+import { RiddleBasicInformationForm } from './RiddleBasicInformationForm';
+import { RiddleQuestionForm } from './RiddleQuestionForm';
+import { RiddleShareForm } from './RiddleShareForm';
+
+const steps = ['Basic information', 'Questions', 'Sharing options'];
+
+export const CreateRiddleForm = () => {
+	const [activeStep, setActiveStep] = useState(0);
+
+	const handleNext = () => {
+		setActiveStep(prevActiveStep => prevActiveStep + 1);
+	};
+
+	const handleBack = () => {
+		setActiveStep(prevActiveStep => prevActiveStep - 1);
+	};
+
+	const formContext = useForm<RiddleUpsertDetail>({
```

Comment:
For update, these values will have to replaced. Otherwise this form can be used.